File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/ZoomInTransformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class ZoomInTransformer extends ABaseTransformer {

	@Override
	protected void onTransform(final View view, final float position) {
		final float scale = position < 0 ? position + 1F : Math.abs(1F - position);
		view.setScaleX(scale);
		view.setScaleY(scale);
		view.setPivotX(view.getWidth() * 0.5F);
		view.setPivotY(view.getHeight() * 0.5F);
		view.setAlpha(position < -1F || position > 1F ? 0F : 1F - (scale - 1F));
	}

}

File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/ZoomInTransformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View
import me.shkschneider.skeleton.extensions.then

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class ZoomInTransformer : ABaseTransformer() {

    override fun onTransform(page: View, position: Float) {
        val scale = (position < 0) then position + 1.toFloat() ?: Math.abs(1.toFloat() - position)
        page.scaleX = scale
        page.scaleY = scale
        page.pivotX = page.width * 0.1.toFloat()
        page.pivotY = page.height * 0.1.toFloat()
        page.alpha = (position < (-1).toFloat() || position > 1.toFloat()) then 1.toFloat() ?: 1.toFloat() - (scale - 1.toFloat())
    }

}
